src/autogluon/cloud/scripts/ray_scripts/train.py

```diff
@@ -0,0 +1,106 @@
+import argparse
+import os
+import shutil
+import time
+from datetime import datetime, timezone
+from typing import Optional
+
+import boto3
+import ray
+import yaml
+
+from autogluon.common.utils.s3_utils import s3_path_to_bucket_prefix
+from autogluon.tabular import TabularDataset, TabularPredictor
+
+
+def wait_for_nodes_to_be_ready():
+    expected_num_nodes = int(os.environ.get("AG_NUM_NODES"))
+    ray.init("auto")
+    print("Waiting for worker nodes to be ready")
+    while len(ray.nodes()) < expected_num_nodes:
+        time.sleep(5)
+    ray.shutdown()
+    print("All nodes ready")
+
+
+def get_utc_timestamp_now():
+    return datetime.now(timezone.utc).strftime("%Y%m%d%H%M%S")
+
+
+def tear_down_cluster(cluster_config_file):
+    import subprocess
+
+    print("Will tear down the cluster in 10 secs")
+    time.sleep(10)
+    cmd = f"ray stop --force; ray down {cluster_config_file} -y"
+    subprocess.Popen(cmd, shell=True, preexec_fn=os.setpgrp)  # Avoid being terminated because ray runtime is down
+
+
+def upload_file(file_name: str, bucket: str, prefix: Optional[str] = None):
+    """
+    Upload a file to an S3 bucket
+
+    Parameters
+    ----------
+    file_name: str,
+        File to upload
+    bucket: str,
+        Bucket to upload to
+    prefix: Optional[str], default = None
+        S3 prefix. If not specified then will upload to the root of the bucket
+    """
+    object_name = os.path.basename(file_name)
+    if prefix is not None and len(prefix) == 0:
+        prefix = None
+    if prefix is not None:
+        object_name = prefix + "/" + object_name
+
+    # Upload the file
+    s3_client = boto3.client("s3")
+    s3_client.upload_file(file_name, bucket, object_name)
+
+
+if __name__ == "__main__":
```

Comment:
[minor suggestion] we can use [typer](https://typer.tiangolo.com/) to simplify this significantly

Reply:
This is a good point. Though the issue here is that this script will be ran in the DLC container, meaning `typer` has to be a dependency of AG. I think it's probably not necessary to introduce such a dependency when AG itself doesn't need it